### ISSUE TITLE
UIResourceChangeRegistry schedules unneccessary Builds

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#217]
UIResourceChangeRegistry schedules unneccessary builds if xtend gets activated lazy

(1) create a mydsl project
(2) start runtime eclipse with new workspace
(3) create a new java project with mydsl file and `Hello World!`
(4) save and restart eclipse
(5) Open inside the restarted Eclipse the "Open Model Element Dialog". This Triggers Xtend to be activated and UIResourceChangeRegistry scheduling a  forgetBuildState build which will lead to a subsequent full build

**Comment by cdietrich** [comment]
@svenefftinge @szarnekow do you know what this class is used for?

**Comment by szarnekow** [comment]
Yes, it is used for Active Annotations. If the annotation processor reads other files, it is safe to assume that it depends on the contents of the file / folder. Thus a listener is registered that'll trigger a build if the resource changes. The dependency tree on resources is persisted by means of the UIResourceChangeRegistry.

**Comment by cdietrich** [comment]
maybe we could have a participant api for `org.eclipse.xtext.builder.builderState.AbstractBuilderState.load()`
that might load/create the file without even activating xtend

**Comment by szarnekow** [comment]
I like the idea of an IBuilderStateParticipant that can be registered by an extension point. That would allow plugin authors to load / save state in the lifecycle of the index data and eliminate the crude build trigger as part e.g. of the Xtend initialization.

**Comment by cdietrich** [comment]
see https://bugs.eclipse.org/bugs/show_bug.cgi?id=449668 as well
